tests/unit/machine/glm_unittest.cc

```diff
@@ -0,0 +1,88 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ * 
+ * Both the documentation and the code is heavily inspired by pyGLMnet.: https://github.com/glm-tools/pyglmnet/
+ *
+ * Author: Tej Sukhatme
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/machine/GLM.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/preprocessor/PruneVarSubMean.h>
+#include <shogun/preprocessor/NormOne.h>
+#include <shogun/mathematics/UniformRealDistribution.h>
+#include <shogun/mathematics/RandomNamespace.h>
+
+#include <random>
+
+using namespace shogun;
+
+//Generate the Data that N is greater than D
+void generate_train_data(SGMatrix<float64_t> &data, SGVector<float64_t> &lab)
+{
+
+	lab = SGVector<float64_t>({7.23514031, 7.23514031, 7.23514031, 7.23514031, 7.23514031});
+	data = SGMatrix<float64_t>({{ 0.71307143, -0.67054885, -0.24406853},
+								{-0.79774475, -1.65627891,  0.95675428},
+								{ 0.96709333,  1.81672959,  0.20911922},
+								{ 2.1912712,   0.23820139,  1.07501177},
+								{-0.58427793, -0.61855905,  1.27687684}});
+}
```

Comment:
it would be nicer if you returned something like `std::tuple<SGMatrix<float64_t>, SGVector<float64_t>>` and then in the function call just have `auto [Xtrain, ytrain] = generate_train_data();`

Reply:
Yes, okay, I'll do this.